### PR TITLE
feat: clean to_eth_transaction_receipt

### PR DESCRIPTION
**Pull request by greged93**
Time spent on this PR: 0.1 day

Resolves: #695 

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?
Replace events parsing with a call to `create_address` from reth_primitives.

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

- - -
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot-rpc/701)

### CODE REVIEW

File: src/models/transaction_receipt.rs [hunk 1]
```diff
@@ -65,24 +64,7 @@ impl StarknetTransactionReceipt {
...
-                                        event.data.first().ok_or(DataDecodingError::InvalidReturnArrayLength {
-                                            entrypoint: "deployment".into(),
-                                            expected: 1,
-                                            actual: 0,
-                                        })?;
-
-                                    let evm_address = Felt252Wrapper::from(*evm_address);
-                                    Some(evm_address.try_into()?)
-                                }
+                                None => Some(create_address(eth_tx.from, eth_tx.nonce.as_u64())),
```
Comment (Eikix) [inline src/models/transaction_receipt.rs:67]: Can we add an invariant for now?
For instance, check that there is a proxy at the get_class_hash_at(compute_starknet_address(evm_address)) == PROXY_CLASS_HASH

Comment (greged93) [reply src/models/transaction_receipt.rs:67]: but this will slow down the execution because we have to make an extra call a provider :/

Comment (Eikix) [reply src/models/transaction_receipt.rs:67]: but in the early days we need to make sure that there is a contract deployed there, no?
what is the "fastest" way to check this invariant?

Comment (Eikix) [reply src/models/transaction_receipt.rs:67]: Alright!

**Review by Eikix — APPROVED** [review]
lgtm

File: src/models/transaction_receipt.rs (hunk 1 above)
Comment (greged93) [reply src/models/transaction_receipt.rs:67]: I think the only way is indeed to call the provider. But would prefer to avoid this.

**Pull request merged**